app/mailers/team_mailer.rb

```diff
@@ -1,14 +1,13 @@
 # An ApplicationMailer subclass that is responsible for informing users about team-related changes via email
 class TeamMailer < ApplicationMailer
-  after_action :prevent_delivery_to_unsubscribed_users
```

Comment:
@conradlempert Any reason why you replaced this hook by manually calling `prevent_delivery_to_team_unsubscribed_users` at the end of the two methods?

Reply:
Ja, das ist etwas tricky. Ich wollte die `prevent`-Methoden in die Oberklasse `Application-Mailer` legen, damit diese aus allen Mailern aufgerufen werden können, und nicht doppelt gecodet sind. In der Methode brauche ich ein `User`-Objekt, um die Einstellung zu checken. Wenn ich das einfach durch `@user`aufrufe, beschwert sich **code climate**, dass dieses Attribut zu diesem Zeitpunkt gar nicht definiert sein könnte. Also muss ich `user` als Parameter an die `prevent`-Methode übergeben. In der hook kann ich keinen Parameter übergeben. Also muss ich die `prevent`-Methode direkt aus den Methoden aufrufen. 

Falls das problematisch ist, kannst du mir gerne ein nices ruby-pattern sagen, mit dem das schöner geht. Mir ist leider keins eingefallen 😝

Reply:
Ah, du hast recht – dass die Methode ja jetzt parametrisiert ist und dadurch nicht mehr per hook aufgerufen werden kann, hatte ich übersehen. Alles klar, dann kann das so bleiben. 👍 Danke für die Klärung!